apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/trace/TraceSegment.java

```diff
@@ -121,16 +116,19 @@ public String getTraceSegmentId() {
         return traceSegmentId;
     }
 
-    public boolean hasRef() {
-        return !(refs == null || refs.size() == 0);
-    }
-
+    /**
+     * For testing use only.
+     */
     public List<TraceSegmentRef> getRefs() {
```

Comment:
Why this API doesn't change? You are optimizing the implementation, you should clear the legacy codes once for all. Otherwise, people are going to be confused when reading this next time.

Reply:
done.